File: SIMO.c
#include "SNN.h"

void elementry_multiplication(double *weight, double input, double *output_vector, int LABLES_COUNT) {
	// EM_result = W1*i/p + W2*i/p + W3*i/p + ....... +  Wn*i/p (W = weight, i/p = input;
	for (int i = 0; i < LABLES_COUNT; i++) {
		output_vector[i] = input * weight[i];
	}
}

void SIMO(double* weight, double input, double* output_vector, int LABLES_COUNT) {
	// single input single output NN;
	elementry_multiplication(weight, input, output_vector, LABLES_COUNT);
}

File: algo/simo/simo.py
def simo(weights, input):
    '''
    Args:
        input: input value in number.
        weight: input value in array of number.
    Returns:
            output return in array of number.
    Raises:
        ValueError: datatype error.
        ValueError: invalid weights error.
    '''

    if not isinstance(weights, list):
        raise ValueError('weight must be an list of float values')

    output = []

    for _weight in weights:
        if not isinstance(_weight, float) or not isinstance(input, float):
            raise ValueError('input and weight must be float value')
        output.append(input * _weight)
    return output

    